### ISSUE TITLE
Broken refman.tex with SHOW_FILES=NO and doxygen groups

### ISSUE THREAD

**Issue by ArnauPrat** [doxygen/doxygen#7038]
Hi all,

We have a project where we use doxygen to generate the documentation. Such project sets the option SHOW_FILES to NO. The project makes extensive use to the **\addtogroup** feature of doxygen. 

When generating the latex files with doxygen, the generated refman.tex file contains an **\input{files}** statement, but the **files.tex** file is not generated by doxygen, thus breaking the compilation when executing make to generate the PDF.

The first version where this problem shows up is 1.8.12. I am currently using 1.8.15. Diving into the code I was able to find a point where 1.8.11 (which works) and 1.8.12 divert:

1.8.12
https://github.com/doxygen/doxygen/blob/4951df8d0d0acf843b4147136f945504b96536e7/src/filedef.cpp#L1891

1.8.11
https://github.com/doxygen/doxygen/blob/a6d4f4df45febe588c38de37641513fd576b998f/src/filedef.cpp#L1880

where you can see the return value differs at the end, where in version >1.8.12 it checks whether the definition is linkable from a group.

Function isLinkableInProject is called from fileVisibleInIndex in 
https://github.com/doxygen/doxygen/blob/4951df8d0d0acf843b4147136f945504b96536e7/src/util.cpp#L8295
which in turn is called from countFiles in index.cpp https://github.com/doxygen/doxygen/blob/4951df8d0d0acf843b4147136f945504b96536e7/src/index.cpp#L1076.

I've observed that when the number of counted documents is 0, the statement **\input{files}** is not generated https://github.com/doxygen/doxygen/blob/4951df8d0d0acf843b4147136f945504b96536e7/src/index.cpp#L4017

I attach  a small example which reproduces the issue. 
[test.zip](https://github.com/doxygen/doxygen/files/3262638/test.zip)

The output of ```doxygen -x Doxygen.test``` is:
```
# Difference with default Doxyfile 1.8.15
SHOW_FILES             = NO
```

**Comment by albert-github** [comment]
Can you please cleanup the question a bit, i.e. don't include the entire Doxyfile in the issue, but only show the differences with the default Doxyfile (as of 1.8.15 with it can be generated with `doxygen -x`). In the current state it is very hard to read (due to its length).
So place the source code plus the Doxyfile in a tar or zip file and attach this to the issue, this makes it also easier to reproduce the problem as no cut / paste is necessary.

**Comment by ArnauPrat** [comment]
Hi @albert-github,
I've updated the text as you requested.
Thanks

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7039

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
